src/app/course/event/event-row/event-row.component.ts

```diff
@@ -33,4 +43,37 @@ export class EventRowComponent implements OnInit {
             this.reload.emit(true)
         })
     }
+
+    /**
+     * Delete an event from the course-list.
+     */
+    async deleteEvent(eventId: number) {
+        this.courseEventService.deleteCourseEvent(eventId).subscribe()
+
+        this.notificationsService
+            .show('The event has been deleted successfully.', {
+                status: TuiNotification.Success
+            }).subscribe()
+        this.router.navigateByUrl('/RefreshComponent', {skipLocationChange: true}).then(() => {
```

Comment:
what is this `RefreshComponent`? and why do we have 2 navigations in side each other?

Reply:
I was trying to refresh the page after an event was deleted and this stack overflow was what I pulled from while trying to at the top comment
https://stackoverflow.com/questions/47813927/how-to-refresh-a-component-in-angular
After I got it working I tried it just with navigating to the same page but it didn't seem to do anything when i tested it

Reply:
window.location.reload() doesn't work, it pulls up the skeleton again but the deleted event still appears after. location.reload has some funky use cases where refreshing the page before you delete sometimes causes it to fail, I've taken a video of it. I believe RefreshComponent doesn't actually exist, I can't find it in any documentation and when I swapped it out for course/1000 in testing it still worked. I imagine what this is doing is saying navigate away from this page but don't show it and then immediately come back forcing the page to effectively do a total reset.

Reply:
We need a better solution, let's discuss it during the meeting.